test/integration/common/StudioActions.kt

```diff
@@ -170,49 +168,67 @@ object StudioActions {
         StudioState.client.tryUpdateTransactionType(TypeDBTransaction.Type.WRITE)
         delayAndRecompose(composeRule, Delays.NETWORK_IO)
 
+        waitUntilNodeWithTextIsClickable(composeRule, Label.SCHEMA.lowercase())
         clickText(composeRule, Label.SCHEMA.lowercase())
+        waitUntilNodeWithTextIsClickable(composeRule, Label.WRITE.lowercase())
         clickText(composeRule, Label.WRITE.lowercase())
 
-        StudioState.project.current!!.directory.entries.find { it.name == schemaFileName }!!.asFile().tryOpen()
+        waitUntilAssertionIsTrue(composeRule) {
+            StudioState.project.current!!.directory.entries.find { it.name == schemaFileName }!!.asFile().tryOpen()
+        }
+
+        delayAndRecompose(composeRule, Delays.FILE_IO)
```

Comment:
Hm. Shouldn't the `waitUntil`-based approach eliminate the need for this delay?

Looks like the same goes for the one on L188, and there may be other examples.

Reply:
Sometimes we wait for a static amount of time in addition to the `waitUntil` as there are certain things we cannot measure. Under certain conditions, we can't tell whether the entire contents of a file is loaded and may accidentally end up committing half a file. This will fail our data validation step later on.

Reply:
Makes sense. In that case, let's continue to delay here, but "dress it up" with a thin wrapper method; something like "waitForFileToBeFullyLoaded", but ideally a shorter name.

That opens up the way for this method to be implemented cleanly in the future, while helping to clarify the intent of the code as-is.

Reply:
Done!